design/hardware-classification-controller/support-for-new-parameters-hwcc-DiskAndNIC.md

```diff
@@ -11,19 +11,25 @@ classify a baremetal host on more possible configurations.
 
 ## Motivation
 
-We have hardware classification controller which validates and
-classifies the hosts based on user hardware requirements.
-We would like to add more parameters which helps user to
+The current HWCC allows user to identify the ideal host by
+specifying minimal parameters.
+We would like to add Disk and NIC parameters which helps user to
 identify the ideal host. This feature helps user to avoid runtime failures
 and also increases the performance for workload deployments.
 
 ### Goals
 
 To make classification more useful by incorporating additional parameters
-to get the matching host. It will help user to specify more rules by
-introducing new parameters.
+to get the matching host.
+Introduce `diskSelector` under Disk which provide the `hctl` pattern and
+`rotational` to filter the disk as per user requirement.
```

Comment:
Are we only adding `hctl` and `rotational` or are we going to make it mirror the structure of the `rootDeviceHints` for consistency?

Reply:
Yes we are adding only those two. We can identify the disk type using `hctl` and `rotational` parameters, so there is no need of other parameters from `rootDeviceHints`.

Reply:
That choice solves the immediate use case, but gives us an API that isn't consistent. I think that's probably OK, but we should document why we made the choice and that we could extend the fields to match in the future. That information could go in the "alternatives" section at the bottom of the document.